### PR TITLE
Add OCI support in experimental flow

### PR DESCRIPTION
**Pull request by agatakrajewska**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Adding OCI image scanning to the experimental flow.

#### Where should the reviewer start?
Commit by commit please :-) 

#### What are the relevant tickets?
[RUN-952](https://snyksec.atlassian.net/browse/RUN-952)

### CODE REVIEW

File: lib/experimental.ts [hunk 1]
```diff
@@ -18,8 +18,9 @@ export async function experimentalAnalysis(
   const imageType = getImageType(targetImage);
   switch (imageType) {
     case ImageType.DockerArchive:
-      return dockerArchive(targetImage, dockerfileAnalysis);
-
+      return localArchive(targetImage, imageType, dockerfileAnalysis);
+    case ImageType.OciArchive:
+      return localArchive(targetImage, imageType, undefined);
```
Comment (ivanstanev) [inline lib/experimental.ts:23]: Should we pass in here the dockerfileAnalysis too?

File: test/lib/image-type.test.ts [hunk 2]
```diff
@@ -15,19 +15,20 @@ test("getImageType() returns the expected transports", (t) => {
...
-test("getDockerArchivePath() returns the expected results", (t) => {
+test("getArchivePath() returns the expected results", (t) => {
   t.plan(2);
 
   t.same(
-    getDockerArchivePath("docker-archive:/tmp/nginx.tar"),
+    getArchivePath("docker-archive:/tmp/nginx.tar"),
     "/tmp/nginx.tar",
     "returns extracted path",
   );
```
Comment (ivanstanev) [inline test/lib/image-type.test.ts:32]: Could we also add a test with `getArchivePath("oci-archive:/tmp/nginx.tar"),`

File: test/lib/image-type.test.ts [hunk 3]
```diff
@@ -15,19 +15,20 @@ test("getImageType() returns the expected transports", (t) => {
...
-  t.same(
-    getDockerArchivePath("bad-path"),
-    "",
-    "does not handle errors and returns empty path",
-  );
+  t.throws(() => getArchivePath("bad-path"), "throws when provided bad path");
```
Comment (ivanstanev) [inline test/lib/image-type.test.ts:33]: Wondering if we should also validate the thrown error, e.g. `Error('The provided archive path is missing image specific prefix, eg."docker-archive:" or "oci-archive:"')`

File: lib/experimental.ts [hunk 4]
```diff
@@ -92,11 +95,12 @@ async function getStaticAnalysisResult(
   targetImage: string,
   archivePath: string,
   dockerfileAnalysis: DockerFileAnalysis | undefined,
+  imageType?: ImageType | undefined,
```
Comment (ivanstanev) [inline lib/experimental.ts:98]: Why undefined here, the `localArchive()` passes in `ImageType` as defined

File: lib/experimental.ts (hunk 1 above)
Comment (agatakrajewska) [reply lib/experimental.ts:23]: Omg, I think I got this wrong? Can we have a dockerfile analysis here?

File: test/lib/image-type.test.ts (hunk 3 above)
Comment (agatakrajewska) [reply test/lib/image-type.test.ts:33]: We can totally.

File: lib/experimental.ts (hunk 4 above)
Comment (agatakrajewska) [reply lib/experimental.ts:98]: But in `distroless` flow we pass `undefined`: https://github.com/snyk/snyk-docker-plugin/pull/225/files/549edc58ae6eae664023543fdcd9d6cfb0feeb94#diff-a29e3cbf53069cd5ba49e11beb3700efL81

Comment (agatakrajewska) [reply lib/experimental.ts]: Actually, I looked into it and I introduced a bug! So before in distroless flow we were assuming `docker-archive`.  I have fixed that now and set `ImageType.DockerArchive` in `distroless()` function, so we are still assuming if `docker-archive:` or `oci-archive:` the default is docker archive.

File: lib/experimental.ts (hunk 1 above)
Comment (ivanstanev) [reply lib/experimental.ts:23]: The analysis was already done earlier (first step of calling `inspect()`), just not sure what it's used for but I think it should be included in the function call!

File: lib/experimental.ts [hunk 5]
```diff
@@ -82,6 +85,7 @@ export async function distroless(
       targetImage,
       archiveFullPath,
       dockerfileAnalysis,
+      ImageType.DockerArchive,
```
Comment (ivanstanev) [inline lib/experimental.ts:88]: Nice!

**Pull request merged**